failed syz-cluster/controller test

https://github.com/google/syzkaller/actions/runs/23444896543/job/68205773121

```
panic: test timed out after 10m0s
	running tests:
		TestProcessor (10m0s)

goroutine 6035 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:2802 +0x34b
created by time.goFunc
	/usr/local/go/src/time/sleep.go:215 +0x2d

goroutine 1 [chan receive, 10 minutes]:
testing.(*T).Run(0x193beff22248, {0x2504bae?, 0x193bf0131b30?}, 0x2684d98)
	/usr/local/go/src/testing/testing.go:2109 +0x4e5
testing.runTests.func1(0x193beff22248)
	/usr/local/go/src/testing/testing.go:2585 +0x37
testing.tRunner(0x193beff22248, 0x193bf0131c58)
	/usr/local/go/src/testing/testing.go:2036 +0xea
testing.runTests({0x25235b1, 0x1b}, {0x2565b6e, 0x32}, 0x193bf00d8660, {0x3e56600, 0x2, 0x2}, {0xc26875494fc2ae74, 0x8bb8011a9d, ...})
	/usr/local/go/src/testing/testing.go:2583 +0x505
testing.(*M).Run(0x193befd02e60)
	/usr/local/go/src/testing/testing.go:2443 +0x6ac
main.main()
	_testmain.go:58 +0x9b

goroutine 7 [select]:
go.opencensus.io/stats/view.(*worker).start(0x193befe6cb80)
	/syzkaller/.cache/gomod/go.opencensus.io@v0.24.0/stats/view/worker.go:292 +0xa5
created by go.opencensus.io/stats/view.init.0 in goroutine 1
	/syzkaller/.cache/gomod/go.opencensus.io@v0.24.0/stats/view/worker.go:34 +0x8d

goroutine 66 [sync.WaitGroup.Wait, 9 minutes]:
sync.runtime_SemacquireWaitGroup(0x193bf01cdc38?, 0xcf?)
	/usr/local/go/src/runtime/sema.go:114 +0x2e
sync.(*WaitGroup).Wait(0x193bf0488a10)
	/usr/local/go/src/sync/waitgroup.go:206 +0x85
runtime.Goexit()
	/usr/local/go/src/runtime/panic.go:694 +0x5e
testing.(*common).FailNow(0x193beff22488)
	/usr/local/go/src/testing/testing.go:1022 +0x4a
testing.(*common).Fatalf(0x193beff22488, {0x252fef8?, 0x0?}, {0x193bf01cddf8?, 0x0?, 0x0?})
	/usr/local/go/src/testing/testing.go:1228 +0x59
github.com/google/syzkaller/syz-cluster/controller.awaitFinishedSessions(0x193beff22488, 0x193bf0620f10, 0xa)
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/controller/processor_test.go:151 +0x226
github.com/google/syzkaller/syz-cluster/controller.TestProcessor(0x193beff22488)
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/controller/processor_test.go:79 +0x693
testing.tRunner(0x193beff22488, 0x2684d98)
	/usr/local/go/src/testing/testing.go:2036 +0xea
created by testing.(*T).Run in goroutine 1
	/usr/local/go/src/testing/testing.go:2101 +0x4c5

goroutine 67 [IO wait, 10 minutes]:
internal/poll.runtime_pollWait(0x7f0ed0f52c00, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0x193befcacae0?, 0x193bf01bc000?, 0x1)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x193befcacae0, {0x193bf01bc000, 0x2000, 0x2000})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
os.(*File).read(...)
	/usr/local/go/src/os/file_posix.go:29
os.(*File).Read(0x193befb5c3f0, {0x193bf01bc000?, 0x1d?, 0x0?})
	/usr/local/go/src/os/file.go:144 +0x4f
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf0487170, {0x26cdcd8, 0x193beff605f0})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 107 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf04871a0, {0x26cdcd8, 0x193beff607d0})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 108 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf04871d0, {0x26cdcd8, 0x193beff60820})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 171 [IO wait]:
internal/poll.runtime_pollWait(0x7f0ed0f52a00, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0x193bf030d180?, 0x193bf02d6000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x193bf030d180, {0x193bf02d6000, 0x8000, 0x8000})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
net.(*netFD).Read(0x193bf030d180, {0x193bf02d6000?, 0x193bf0492fa0?, 0x30?})
	/usr/local/go/src/net/fd_posix.go:68 +0x25
net.(*conn).Read(0x193bf0102928, {0x193bf02d6000?, 0x48d64c?, 0x193bf02a0d48?})
	/usr/local/go/src/net/net.go:196 +0x45
bufio.(*Reader).Read(0x193bf0484f00, {0x193befe18ba4, 0x9, 0x22c32e0?})
	/usr/local/go/src/bufio/bufio.go:245 +0x197
io.ReadAtLeast({0x2696ce0, 0x193bf0484f00}, {0x193befe18ba4, 0x9, 0x9}, 0x9)
	/usr/local/go/src/io/io.go:335 +0x8e
io.ReadFull(...)
	/usr/local/go/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x193befe18ba4, 0x9, 0x193b00000000?}, {0x2696ce0?, 0x193bf0484f00?})
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:250 +0x65
golang.org/x/net/http2.(*Framer).ReadFrameHeader(0x193befe18b60)
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:513 +0x6b
google.golang.org/grpc/internal/transport.(*framer).readFrame(0x193bf0482380)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http_util.go:480 +0x45
google.golang.org/grpc/internal/transport.(*http2Client).reader(0x193bf0429688, 0x193bf03009a0)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1658 +0x1ba
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 61
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:411 +0x1e13

goroutine 166 [select]:
google.golang.org/grpc/internal/transport.(*http2Client).keepalive(0x193bf0429448)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1732 +0x185
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 105
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:401 +0x1d9d

goroutine 112 [chan receive, 9 minutes]:
cloud.google.com/go/spanner.(*sessionManager).createMultiplexedSession(0x193befe18540)
	/syzkaller/.cache/gomod/cloud.google.com/go/spanner@v1.88.0/session.go:342 +0x5a
created by cloud.google.com/go/spanner.newSessionManager in goroutine 66
	/syzkaller/.cache/gomod/cloud.google.com/go/spanner@v1.88.0/session.go:288 +0x7c5

goroutine 114 [select, 9 minutes]:
cloud.google.com/go/spanner.(*sessionManager).multiplexSessionWorker(0x193befe18540)
	/syzkaller/.cache/gomod/cloud.google.com/go/spanner@v1.88.0/session.go:572 +0x25b
created by cloud.google.com/go/spanner.newSessionManager in goroutine 66
	/syzkaller/.cache/gomod/cloud.google.com/go/spanner@v1.88.0/session.go:304 +0x8d9

goroutine 115 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf0487b00, {0x26cdcd8, 0x193befeea780})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 116 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf0487b30, {0x26cdcd8, 0x193befeea7d0})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 117 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf0487b60, {0x26cdcd8, 0x193befeea820})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 167 [IO wait]:
internal/poll.runtime_pollWait(0x7f0ed0f52600, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0x193bf030d800?, 0x193bf0258000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x193bf030d800, {0x193bf0258000, 0x8000, 0x8000})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
net.(*netFD).Read(0x193bf030d800, {0x193bf0258000?, 0x193bf0492e60?, 0x30?})
	/usr/local/go/src/net/fd_posix.go:68 +0x25
net.(*conn).Read(0x193bf01028f0, {0x193bf0258000?, 0x48d64c?, 0x193bf0418d48?})
	/usr/local/go/src/net/net.go:196 +0x45
bufio.(*Reader).Read(0x193bf0484de0, {0x193befe18ac4, 0x9, 0x22c32e0?})
	/usr/local/go/src/bufio/bufio.go:245 +0x197
io.ReadAtLeast({0x2696ce0, 0x193bf0484de0}, {0x193befe18ac4, 0x9, 0x9}, 0x9)
	/usr/local/go/src/io/io.go:335 +0x8e
io.ReadFull(...)
	/usr/local/go/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x193befe18ac4, 0x9, 0x193b00000000?}, {0x2696ce0?, 0x193bf0484de0?})
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:250 +0x65
golang.org/x/net/http2.(*Framer).ReadFrameHeader(0x193befe18a80)
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:513 +0x6b
google.golang.org/grpc/internal/transport.(*framer).readFrame(0x193bf0482280)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http_util.go:480 +0x45
google.golang.org/grpc/internal/transport.(*http2Client).reader(0x193bf0429448, 0x193bf0300850)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1658 +0x1ba
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 105
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:411 +0x1e13

goroutine 162 [select]:
google.golang.org/grpc/internal/transport.(*http2Client).keepalive(0x193bf0429208)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1732 +0x185
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 111
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:401 +0x1d9d

goroutine 121 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf0420db0, {0x26cdcd8, 0x193befeeb270})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 122 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf0420de0, {0x26cdcd8, 0x193befeeb2c0})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 123 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf0420e10, {0x26cdcd8, 0x193befeeb310})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 163 [IO wait]:
internal/poll.runtime_pollWait(0x7f0ed0f52400, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0x193bf030de00?, 0x193bf0248000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x193bf030de00, {0x193bf0248000, 0x8000, 0x8000})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
net.(*netFD).Read(0x193bf030de00, {0x193bf0248000?, 0x193bf0492d20?, 0x30?})
	/usr/local/go/src/net/fd_posix.go:68 +0x25
net.(*conn).Read(0x193bf01028b8, {0x193bf0248000?, 0x48d64c?, 0x193befc8fd48?})
	/usr/local/go/src/net/net.go:196 +0x45
bufio.(*Reader).Read(0x193bf0484cc0, {0x193befe189e4, 0x9, 0x22c32e0?})
	/usr/local/go/src/bufio/bufio.go:245 +0x197
io.ReadAtLeast({0x2696ce0, 0x193bf0484cc0}, {0x193befe189e4, 0x9, 0x9}, 0x9)
	/usr/local/go/src/io/io.go:335 +0x8e
io.ReadFull(...)
	/usr/local/go/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x193befe189e4, 0x9, 0x193b00000000?}, {0x2696ce0?, 0x193bf0484cc0?})
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:250 +0x65
golang.org/x/net/http2.(*Framer).ReadFrameHeader(0x193befe189a0)
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:513 +0x6b
google.golang.org/grpc/internal/transport.(*framer).readFrame(0x193bf0482180)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http_util.go:480 +0x45
google.golang.org/grpc/internal/transport.(*http2Client).reader(0x193bf0429208, 0x193bf0300700)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1658 +0x1ba
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 111
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:411 +0x1e13

goroutine 159 [IO wait, 9 minutes]:
internal/poll.runtime_pollWait(0x7f0ed0f52200, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0x193befe6cc80?, 0x193bf022a000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x193befe6cc80, {0x193bf022a000, 0x8000, 0x8000})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
net.(*netFD).Read(0x193befe6cc80, {0x193bf022a000?, 0x7f0f17b59458?, 0x10?})
	/usr/local/go/src/net/fd_posix.go:68 +0x25
net.(*conn).Read(0x193bf0102880, {0x193bf022a000?, 0x193befc8dd20?, 0x41dfff?})
	/usr/local/go/src/net/net.go:196 +0x45
bufio.(*Reader).Read(0x193bf0484ba0, {0x193befe18904, 0x9, 0x4?})
	/usr/local/go/src/bufio/bufio.go:245 +0x197
io.ReadAtLeast({0x2696ce0, 0x193bf0484ba0}, {0x193befe18904, 0x9, 0x9}, 0x9)
	/usr/local/go/src/io/io.go:335 +0x8e
io.ReadFull(...)
	/usr/local/go/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x193befe18904, 0x9, 0x193befc8ddef?}, {0x2696ce0?, 0x193bf0484ba0?})
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:250 +0x65
golang.org/x/net/http2.(*Framer).ReadFrameHeader(0x193befe188c0)
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:513 +0x6b
google.golang.org/grpc/internal/transport.(*framer).readFrame(0x193bf0482080)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http_util.go:480 +0x45
google.golang.org/grpc/internal/transport.(*http2Client).reader(0x193bf0428fc8, 0x193bf0300690)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1658 +0x1ba
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 120
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:411 +0x1e13

goroutine 127 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf0421460, {0x26cdcd8, 0x193befeebb80})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 128 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf0421490, {0x26cdcd8, 0x193befeebbd0})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 129 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf04214c0, {0x26cdcd8, 0x193befeebc20})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 160 [select, 9 minutes]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0x193bf012e040, 0x1)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:420 +0x10a
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0x193bf0492c80)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:597 +0x78
google.golang.org/grpc/internal/transport.NewHTTP2Client.func6()
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:469 +0xd2
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 120
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:467 +0x23db

goroutine 49 [select]:
google.golang.org/grpc/internal/transport.(*http2Client).keepalive(0x193befba26c8)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1732 +0x185
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 99
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:401 +0x1d9d

goroutine 133 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf0421b20, {0x26cdcd8, 0x193beff1a5a0})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 134 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf0421b50, {0x26cdcd8, 0x193beff1a5f0})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 135 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193bf0421b80, {0x26cdcd8, 0x193beff1a640})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 178 [IO wait]:
internal/poll.runtime_pollWait(0x7f0ed0f52800, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0x193bf030d500?, 0x193bf0268000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x193bf030d500, {0x193bf0268000, 0x8000, 0x8000})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
net.(*netFD).Read(0x193bf030d500, {0x193bf0268000?, 0x193bf032abe0?, 0x30?})
	/usr/local/go/src/net/fd_posix.go:68 +0x25
net.(*conn).Read(0x193bf00881a8, {0x193bf0268000?, 0x48d64c?, 0x193bf0419d48?})
	/usr/local/go/src/net/net.go:196 +0x45
bufio.(*Reader).Read(0x193bf03236e0, {0x193bf02ee2e4, 0x9, 0x22c32e0?})
	/usr/local/go/src/bufio/bufio.go:245 +0x197
io.ReadAtLeast({0x2696ce0, 0x193bf03236e0}, {0x193bf02ee2e4, 0x9, 0x9}, 0x9)
	/usr/local/go/src/io/io.go:335 +0x8e
io.ReadFull(...)
	/usr/local/go/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x193bf02ee2e4, 0x9, 0x193b00000000?}, {0x2696ce0?, 0x193bf03236e0?})
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:250 +0x65
golang.org/x/net/http2.(*Framer).ReadFrameHeader(0x193bf02ee2a0)
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:513 +0x6b
google.golang.org/grpc/internal/transport.(*framer).readFrame(0x193beffdd180)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http_util.go:480 +0x45
google.golang.org/grpc/internal/transport.(*http2Client).reader(0x193befba26c8, 0x193bf04427e0)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1658 +0x1ba
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 99
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:411 +0x1e13

goroutine 155 [sync.Cond.Wait, 8 minutes]:
sync.runtime_notifyListWait(0x193befec3e50, 0x0)
	/usr/local/go/src/runtime/sema.go:617 +0x1b3
sync.(*Cond).Wait(0x193befebbf78?)
	/usr/local/go/src/sync/cond.go:71 +0x73
google.golang.org/grpc/internal/transport.(*http2Client).keepalive(0x193bf0428d88)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1766 +0x26f
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 126
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:401 +0x1d9d

goroutine 139 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193befe4c580, {0x26cdcd8, 0x193beff1af00})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 140 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193befe4c5b0, {0x26cdcd8, 0x193beff1af50})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 141 [chan receive, 9 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x193befe4c5e0, {0x26cdcd8, 0x193beff1afa0})
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:88 +0xe5
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 66
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/grpcsync/callback_serializer.go:52 +0x11a

goroutine 156 [IO wait, 9 minutes]:
internal/poll.runtime_pollWait(0x7f0ed0f52000, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0x193befe6cf80?, 0x193bf021a000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x193befe6cf80, {0x193bf021a000, 0x8000, 0x8000})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
net.(*netFD).Read(0x193befe6cf80, {0x193bf021a000?, 0x42b3be?, 0x2000?})
	/usr/local/go/src/net/fd_posix.go:68 +0x25
net.(*conn).Read(0x193bf0102848, {0x193bf021a000?, 0x193befc89d20?, 0x41dfff?})
	/usr/local/go/src/net/net.go:196 +0x45
bufio.(*Reader).Read(0x193bf0484a80, {0x193befe18824, 0x9, 0x4?})
	/usr/local/go/src/bufio/bufio.go:245 +0x197
io.ReadAtLeast({0x2696ce0, 0x193bf0484a80}, {0x193befe18824, 0x9, 0x9}, 0x9)
	/usr/local/go/src/io/io.go:335 +0x8e
io.ReadFull(...)
	/usr/local/go/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x193befe18824, 0x9, 0x193befc89def?}, {0x2696ce0?, 0x193bf0484a80?})
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:250 +0x65
golang.org/x/net/http2.(*Framer).ReadFrameHeader(0x193befe187e0)
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:513 +0x6b
google.golang.org/grpc/internal/transport.(*framer).readFrame(0x193befe6de00)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http_util.go:480 +0x45
google.golang.org/grpc/internal/transport.(*http2Client).reader(0x193bf0428d88, 0x193bf03003f0)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1658 +0x1ba
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 126
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:411 +0x1e13

goroutine 45 [sync.Cond.Wait, 8 minutes]:
sync.runtime_notifyListWait(0x193bf0310fd0, 0x0)
	/usr/local/go/src/runtime/sema.go:617 +0x1b3
sync.(*Cond).Wait(0x193befeb5f78?)
	/usr/local/go/src/sync/cond.go:71 +0x73
google.golang.org/grpc/internal/transport.(*http2Client).keepalive(0x193bf0329208)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1766 +0x26f
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 132
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:401 +0x1d9d

goroutine 145 [chan receive, 9 minutes]:
cloud.google.com/go/spanner.(*sessionManager).createMultiplexedSession(0x193befe18620)
	/syzkaller/.cache/gomod/cloud.google.com/go/spanner@v1.88.0/session.go:342 +0x5a
created by cloud.google.com/go/spanner.newSessionManager in goroutine 66
	/syzkaller/.cache/gomod/cloud.google.com/go/spanner@v1.88.0/session.go:288 +0x7c5

goroutine 147 [select, 9 minutes]:
cloud.google.com/go/spanner.(*sessionManager).multiplexSessionWorker(0x193befe18620)
	/syzkaller/.cache/gomod/cloud.google.com/go/spanner@v1.88.0/session.go:572 +0x25b
created by cloud.google.com/go/spanner.newSessionManager in goroutine 66
	/syzkaller/.cache/gomod/cloud.google.com/go/spanner@v1.88.0/session.go:304 +0x8d9

goroutine 46 [IO wait, 9 minutes]:
internal/poll.runtime_pollWait(0x7f0ed0f51e00, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0x193befe6d280?, 0x193bf020a000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x193befe6d280, {0x193bf020a000, 0x8000, 0x8000})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
net.(*netFD).Read(0x193befe6d280, {0x193bf020a000?, 0x193bf032a640?, 0x30?})
	/usr/local/go/src/net/fd_posix.go:68 +0x25
net.(*conn).Read(0x193bf0088110, {0x193bf020a000?, 0x193befc8ad20?, 0x41dfff?})
	/usr/local/go/src/net/net.go:196 +0x45
bufio.(*Reader).Read(0x193bf0322a80, {0x193bf02ee204, 0x9, 0x4?})
	/usr/local/go/src/bufio/bufio.go:245 +0x197
io.ReadAtLeast({0x2696ce0, 0x193bf0322a80}, {0x193bf02ee204, 0x9, 0x9}, 0x9)
	/usr/local/go/src/io/io.go:335 +0x8e
io.ReadFull(...)
	/usr/local/go/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x193bf02ee204, 0x9, 0x193befc8adef?}, {0x2696ce0?, 0x193bf0322a80?})
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:250 +0x65
golang.org/x/net/http2.(*Framer).ReadFrameHeader(0x193bf02ee1c0)
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:513 +0x6b
google.golang.org/grpc/internal/transport.(*framer).readFrame(0x193beffdca00)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http_util.go:480 +0x45
google.golang.org/grpc/internal/transport.(*http2Client).reader(0x193bf0329208, 0x193bf04425b0)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1658 +0x1ba
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 132
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:411 +0x1e13

goroutine 41 [sync.Cond.Wait, 8 minutes]:
sync.runtime_notifyListWait(0x193bf0310f10, 0x0)
	/usr/local/go/src/runtime/sema.go:617 +0x1b3
sync.(*Cond).Wait(0x193befeb4778?)
	/usr/local/go/src/sync/cond.go:71 +0x73
google.golang.org/grpc/internal/transport.(*http2Client).keepalive(0x193bf0328fc8)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1766 +0x26f
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 144
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:401 +0x1d9d

goroutine 42 [IO wait, 9 minutes]:
internal/poll.runtime_pollWait(0x7f0ecc35a200, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0x193befe6dc80?, 0x193bf01e6000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x193befe6dc80, {0x193bf01e6000, 0x8000, 0x8000})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
net.(*netFD).Read(0x193befe6dc80, {0x193bf01e6000?, 0x193bf032a500?, 0x30?})
	/usr/local/go/src/net/fd_posix.go:68 +0x25
net.(*conn).Read(0x193bf00880c0, {0x193bf01e6000?, 0x193bf029fd20?, 0x41dfff?})
	/usr/local/go/src/net/net.go:196 +0x45
bufio.(*Reader).Read(0x193bf0322960, {0x193bf02ee124, 0x9, 0x4?})
	/usr/local/go/src/bufio/bufio.go:245 +0x197
io.ReadAtLeast({0x2696ce0, 0x193bf0322960}, {0x193bf02ee124, 0x9, 0x9}, 0x9)
	/usr/local/go/src/io/io.go:335 +0x8e
io.ReadFull(...)
	/usr/local/go/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x193bf02ee124, 0x9, 0x193bf029fdef?}, {0x2696ce0?, 0x193bf0322960?})
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:250 +0x65
golang.org/x/net/http2.(*Framer).ReadFrameHeader(0x193bf02ee0e0)
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:513 +0x6b
google.golang.org/grpc/internal/transport.(*framer).readFrame(0x193beffdc800)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http_util.go:480 +0x45
google.golang.org/grpc/internal/transport.(*http2Client).reader(0x193bf0328fc8, 0x193bf0442460)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1658 +0x1ba
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 144
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:411 +0x1e13

goroutine 151 [sync.Cond.Wait, 8 minutes]:
sync.runtime_notifyListWait(0x193befec3d90, 0x0)
	/usr/local/go/src/runtime/sema.go:617 +0x1b3
sync.(*Cond).Wait(0x193befeb9f78?)
	/usr/local/go/src/sync/cond.go:71 +0x73
google.golang.org/grpc/internal/transport.(*http2Client).keepalive(0x193bf0428b48)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1766 +0x26f
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 138
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:401 +0x1d9d

goroutine 152 [IO wait, 9 minutes]:
internal/poll.runtime_pollWait(0x7f0ecc35a400, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0x193befe6d700?, 0x193bf01f0000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x193befe6d700, {0x193bf01f0000, 0x8000, 0x8000})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
net.(*netFD).Read(0x193befe6d700, {0x193bf01f0000?, 0x193bf0492960?, 0x30?})
	/usr/local/go/src/net/fd_posix.go:68 +0x25
net.(*conn).Read(0x193bf0102810, {0x193bf01f0000?, 0x193bf029ed20?, 0x41dfff?})
	/usr/local/go/src/net/net.go:196 +0x45
bufio.(*Reader).Read(0x193bf0484960, {0x193befe18744, 0x9, 0x4?})
	/usr/local/go/src/bufio/bufio.go:245 +0x197
io.ReadAtLeast({0x2696ce0, 0x193bf0484960}, {0x193befe18744, 0x9, 0x9}, 0x9)
	/usr/local/go/src/io/io.go:335 +0x8e
io.ReadFull(...)
	/usr/local/go/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x193befe18744, 0x9, 0x193bf029edef?}, {0x2696ce0?, 0x193bf0484960?})
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:250 +0x65
golang.org/x/net/http2.(*Framer).ReadFrameHeader(0x193befe18700)
	/syzkaller/.cache/gomod/golang.org/x/net@v0.51.0/http2/frame.go:513 +0x6b
google.golang.org/grpc/internal/transport.(*framer).readFrame(0x193befe6dd00)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http_util.go:480 +0x45
google.golang.org/grpc/internal/transport.(*http2Client).reader(0x193bf0428b48, 0x193bf0300070)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:1658 +0x1ba
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 138
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:411 +0x1e13

goroutine 43 [select, 9 minutes]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0x193bf0310ec0, 0x1)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:420 +0x10a
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0x193bf032a5a0)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:597 +0x78
google.golang.org/grpc/internal/transport.NewHTTP2Client.func6()
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:469 +0xd2
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 144
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:467 +0x23db

goroutine 153 [select, 9 minutes]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0x193befec3d40, 0x1)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:420 +0x10a
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0x193bf0492a00)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:597 +0x78
google.golang.org/grpc/internal/transport.NewHTTP2Client.func6()
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:469 +0xd2
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 138
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:467 +0x23db

goroutine 47 [select, 9 minutes]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0x193bf0310f80, 0x1)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:420 +0x10a
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0x193bf032a6e0)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:597 +0x78
google.golang.org/grpc/internal/transport.NewHTTP2Client.func6()
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:469 +0xd2
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 132
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:467 +0x23db

goroutine 157 [select, 9 minutes]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0x193befec3e00, 0x1)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:420 +0x10a
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0x193bf0492b40)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:597 +0x78
google.golang.org/grpc/internal/transport.NewHTTP2Client.func6()
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:469 +0xd2
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 126
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:467 +0x23db

goroutine 164 [select]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0x193bf012e0c0, 0x1)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:420 +0x10a
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0x193bf0492dc0)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:597 +0x78
google.golang.org/grpc/internal/transport.NewHTTP2Client.func6()
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:469 +0xd2
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 111
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:467 +0x23db

goroutine 168 [select]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0x193bf012e180, 0x1)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:420 +0x10a
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0x193bf0492f00)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:597 +0x78
google.golang.org/grpc/internal/transport.NewHTTP2Client.func6()
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:469 +0xd2
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 105
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:467 +0x23db

goroutine 179 [select]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0x193befbaa4c0, 0x1)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:420 +0x10a
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0x193bf032ac80)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:597 +0x78
google.golang.org/grpc/internal/transport.NewHTTP2Client.func6()
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:469 +0xd2
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 99
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:467 +0x23db

goroutine 172 [select]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0x193bf012e240, 0x1)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:420 +0x10a
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0x193bf0493040)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/controlbuf.go:597 +0x78
google.golang.org/grpc/internal/transport.NewHTTP2Client.func6()
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:469 +0xd2
created by google.golang.org/grpc/internal/transport.NewHTTP2Client in goroutine 61
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/http2_client.go:467 +0x23db

goroutine 286 [sync.WaitGroup.Wait, 9 minutes]:
sync.runtime_SemacquireWaitGroup(0x46245f?, 0x70?)
	/usr/local/go/src/runtime/sema.go:114 +0x2e
sync.(*WaitGroup).Wait(0x193bf0815950)
	/usr/local/go/src/sync/waitgroup.go:206 +0x85
github.com/google/syzkaller/syz-cluster/controller.(*SeriesProcessor).Loop(0x193bf06383c0, {0x26cdcd8, 0x193bf0a18f00})
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/controller/processor.go:78 +0x477
github.com/google/syzkaller/syz-cluster/controller.TestProcessor.func2()
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/controller/processor_test.go:65 +0x28
created by github.com/google/syzkaller/syz-cluster/controller.TestProcessor in goroutine 66
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/controller/processor_test.go:64 +0x5e7

goroutine 298 [select]:
github.com/google/syzkaller/syz-cluster/controller.(*SeriesProcessor).handleSession(0x193bf06383c0, {0x26cdcd8, 0x193bf0a18f00}, 0x193bf08a95f0)
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/controller/processor.go:139 +0xf2
github.com/google/syzkaller/syz-cluster/controller.(*SeriesProcessor).seriesRunner.func1()
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/controller/processor.go:128 +0xde
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/syzkaller/.cache/gomod/golang.org/x/sync@v0.19.0/errgroup/errgroup.go:93 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 308
	/syzkaller/.cache/gomod/golang.org/x/sync@v0.19.0/errgroup/errgroup.go:78 +0x95

goroutine 288 [select]:
google.golang.org/grpc/internal/transport.(*ClientStream).waitOnHeader(0x193befd19a00)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/client_stream.go:94 +0x66
google.golang.org/grpc/internal/transport.(*ClientStream).Header(0x193befd19a00)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/internal/transport/client_stream.go:124 +0x27
google.golang.org/grpc.(*clientStream).Header.func1(0x1?)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/stream.go:865 +0x25
google.golang.org/grpc.(*clientStream).withRetry(0x193bf0826900, 0x193befbef818, 0x193befbef808)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/stream.go:840 +0x12d
google.golang.org/grpc.(*clientStream).Header(0x193bf0826900)
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/stream.go:863 +0x56
cloud.google.com/go/spanner.(*grpcSpannerClient).ExecuteStreamingSql(0x193bf0650060, {0x26cdca0, 0x193bf0837770}, 0x193bf08195f0, {0x193beff620d0, 0x1, 0x1})
	/syzkaller/.cache/gomod/cloud.google.com/go/spanner@v1.88.0/grpc_client.go:257 +0x18b
cloud.google.com/go/spanner.(*txReadOnly).query.func2({0x26cdca0, 0x193bf0837770}, {0x0?, 0x2691e80?, 0x220b3a0?}, {0x193beff620d0, 0x1, 0x1})
	/syzkaller/.cache/gomod/cloud.google.com/go/spanner@v1.88.0/transaction.go:737 +0x172
cloud.google.com/go/spanner.(*resumableStreamDecoder).next(0x193bf0375d10, 0x193bf08196c0)
	/syzkaller/.cache/gomod/cloud.google.com/go/spanner@v1.88.0/read.go:570 +0x396
cloud.google.com/go/spanner.(*RowIterator).Next(0x193bf0801e40)
	/syzkaller/.cache/gomod/cloud.google.com/go/spanner@v1.88.0/read.go:182 +0x236
github.com/google/syzkaller/syz-cluster/pkg/db.readRow[...](0x0)
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/pkg/db/spanner.go:239 +0x1f
github.com/google/syzkaller/syz-cluster/pkg/db.readRows[...](0x193bf0801e40)
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/pkg/db/spanner.go:267 +0x73
github.com/google/syzkaller/syz-cluster/pkg/db.readEntities[...]({0x26cdcd8, 0x193bf0a18f00?}, {0x26967a0?, 0x193bf082e540?}, {{0x193bf029ac60?, 0x193befbefdf0?}, 0x193bf0837530?})
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/pkg/db/spanner.go:282 +0xb3
github.com/google/syzkaller/syz-cluster/pkg/db.(*genericEntityOps[...]).readEntities(0x26ebaa0?, {0x26cdcd8, 0x193bf0a18f00}, {{0x193bf029ac60, 0x193bf092ccb0?}, 0x193bf0837530?})
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/pkg/db/spanner.go:368 +0x2b3
github.com/google/syzkaller/syz-cluster/pkg/db.(*SessionRepository).ListWaiting(0x193bf0620f90, {0x26cdcd8, 0x193bf0a18f00}, 0x0, 0x1)
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/pkg/db/session_repo.go:99 +0x27c
github.com/google/syzkaller/syz-cluster/controller.(*SeriesProcessor).streamSeries(0x193bf06383c0, {0x26cdcd8, 0x193bf0a18f00}, 0x193befe7a620)
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/controller/processor.go:95 +0x176
github.com/google/syzkaller/syz-cluster/controller.(*SeriesProcessor).Loop.func2()
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/controller/processor.go:75 +0x66
created by github.com/google/syzkaller/syz-cluster/controller.(*SeriesProcessor).Loop in goroutine 286
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/controller/processor.go:73 +0x455

goroutine 308 [select, 9 minutes]:
github.com/google/syzkaller/syz-cluster/controller.(*SeriesProcessor).seriesRunner(0x193bf06383c0, {0x26cdcd8, 0x193bf0a18f00}, 0x193befe7a620)
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/controller/processor.go:117 +0x166
github.com/google/syzkaller/syz-cluster/controller.(*SeriesProcessor).Loop.func1()
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/controller/processor.go:56 +0x5d
created by github.com/google/syzkaller/syz-cluster/controller.(*SeriesProcessor).Loop in goroutine 286
	/workspace/actions-runner/_work/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/syz-cluster/controller/processor.go:54 +0x16c

goroutine 6013 [select]:
google.golang.org/grpc.newClientStreamWithParams.func4()
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/stream.go:426 +0x8e
created by google.golang.org/grpc.newClientStreamWithParams in goroutine 288
	/syzkaller/.cache/gomod/google.golang.org/grpc@v1.79.1/stream.go:425 +0xeb7
FAIL	github.com/google/syzkaller/syz-cluster/controller	600.150s
```

--- a/syz-cluster/controller/processor_test.go
+++ b/syz-cluster/controller/processor_test.go
@@ -153,33 +153,39 @@ func awaitFinishedSessions(t *testing.T, seriesRepo *db.SeriesRepository, wantFi
 
 type mockedWorkflows struct {
 	workflow.MockService
-	finish  chan struct{}
-	created map[string]struct{}
+	finish   chan struct{}
+	created  map[string]struct{}
+	finished map[string]struct{}
+	mu       sync.Mutex
 }
 
 func newMockedWorkflows() *mockedWorkflows {
 	obj := mockedWorkflows{
-		finish:  make(chan struct{}),
-		created: make(map[string]struct{}),
+		finish:   make(chan struct{}, 100),
+		created:  make(map[string]struct{}),
+		finished: make(map[string]struct{}),
 	}
 	obj.PollDelayValue = time.Millisecond
 	obj.OnStart = func(id string) error {
+		obj.mu.Lock()
+		defer obj.mu.Unlock()
 		obj.created[id] = struct{}{}
 		return nil
 	}
 	obj.OnStatus = func(id string) (workflow.Status, []byte, error) {
-		_, ok := obj.created[id]
-		if !ok {
+		obj.mu.Lock()
+		defer obj.mu.Unlock()
+		if _, ok := obj.created[id]; !ok {
 			return workflow.StatusNotFound, nil, nil
 		}
-		finished := false
+		if _, ok := obj.finished[id]; ok {
+			return workflow.StatusFinished, nil, nil
+		}
 		select {
 		case <-obj.finish:
-			finished = true
-		default:
-		}
-		if finished {
+			obj.finished[id] = struct{}{}
 			return workflow.StatusFinished, nil, nil
+		default:
 		}
 		return workflow.StatusRunning, nil, nil
 	}

